### PR TITLE
e2e: add workflow for e2e module.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

this has bugged me for some time, running `make lint` locally always found something in the `e2e` module to complain about. This workflow adds the linting step for the e2e module and it runs a few of the tests it defines.

closes: #3695 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **New Features**
	- Introduced a GitHub Actions workflow "E2E Module" for automated testing and linting on PRs targeting the e2e module.
- **Refactor**
	- Simplified type conversion and comparison logic in `GenesisTestSuite` within the e2e tests.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
🙏

File: .github/workflows/e2emodule.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: E2E Module
```
Comment (chatton) [inline .github/workflows/e2emodule.yml:1]: Maybe we can rename to E2E Unit tests or something?

**Review by chatton — APPROVED** [review]
LGTM! Thanks for adding this, has bugged me a few times in the past also!

**Comment by chatton** [comment]
I believe this closes #3695 as well.

**Review by damiannolan — APPROVED** [review]
Thank you @DimitrisJim!

**Pull request merged**